src/shared/filesystemUtilities.ts

```diff
@@ -78,14 +78,22 @@ export const makeTemporaryToolkitFolder = async (...relativePathParts: string[])
 }
 
 /**
- * Returns whether or not a directory/file is a contained within a parent directory.
- * Also returns true if parent directory === subdirectory
- * @param parentDirectory Parent directory
- * @param containedPath Path to directory that may or may not be contained within the parentDirectory
+ * Returns `true` if path `p` is a descendant of directory `d` (or if they are
+ * identical).
+ *
+ * Only the logical structure is checked; the paths are not checked for
+ * existence on the filesystem.
+ *
+ * @param d  Path to a directory.
+ * @param p  Path to file or directory to test.
  */
-export function isContainedWithinDirectory(parentDirectory: string, containedPath: string): boolean {
-    const parentDirPieces = parentDirectory.split(path.sep)
-    const containedPathPieces = containedPath.split(path.sep)
+export function isInDirectory(d: string, p: string): boolean {
+    const parentDirPieces = d.split(path.sep)
+    const containedPathPieces = p.split(path.sep)
+    // Remove final empty element(s), if `d` ends with slash(es).
+    while (parentDirPieces.length > 0 && parentDirPieces[parentDirPieces.length - 1] === '') {
+        parentDirPieces.pop()
+    }
 
     return parentDirPieces.every((value, index) => {
```

Comment:
If you've popped all of parentDirPieces, isn't it empty now? Does this give false positives?

Reply:
> If you've popped all of parentDirPieces, isn't it empty now?

That would happen if the directory is `/` or `/////` , which logically contains "everything". Is there some other scenario you are thinking of?

Reply:
Ah I see. Thanks, I misunderstood the code.

Something that is not related to your change, but I just realized from this nth look at the method, we're splitting on `path.sep`, what happens (for example) on windows, when you feed this a "normalized path that uses the / slash?

Reply:
yeah we should probably normalize here. That's why windows CI is failing...

Reply:
Lucky that there is a test using explicit "/" slashes. We should probably add a test that uses explicit "\" slashes to exercise appropriately on the non-windows systems too (though that is a far more uncommon scenario)